src/models/Model.php

```diff
@@ -24,4 +28,31 @@ protected static function getMc(): Memcached {
     }
     return self::$mc;
   }
+
+  protected static function setMCRecords(string $key, mixed $records): void {
+    $mc = self::getMc();
+    $mc->set(
+      static::$MC_KEY.static::$MC_KEYS->get($key),
+      $records,
+      static::$MC_EXPIRE,
+    );
+  }
+
+  protected static function getMCRecords(string $key): mixed {
+    $mc = self::getMc();
+    $mc_result = $mc->get(static::$MC_KEY.static::$MC_KEYS->get($key));
+    /* HH_IGNORE_ERROR[4110] */
+    return $mc_result;
+  }
+
+  public static function invalidateMCRecords(?string $key = null): void {
+    $mc = self::getMc();
+    if (is_null($key)) {
```

Comment:
nit: while `is_null` is equivalent to `!== null`, we've been using the latter so far, so let's stick with it.